File: android_java/Demo_menu/app/src/main/java/com/heyangpeng/demo_menu/FirstFragment.java
package com.heyangpeng.demo_menu;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class FirstFragment extends Fragment {
    private boolean isRefresh;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);//必须在view挂载前添加，一般在onCreate()中。
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        isRefresh = false;
        return inflater.inflate(R.layout.activity_fragment, container, false);

    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        inflater.inflate(R.menu.menu_fragment, menu);//加载自己的menu
    }


    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId()) {
            case R.id.add_item_fragment:
                isRefresh = true;
                getActivity().invalidateOptionsMenu();
                Toast.makeText(getContext(), "REFRESH_FRAGMENT", Toast.LENGTH_SHORT).show();
                return true;
            case R.id.remove_item_fragment:
                Toast.makeText(getContext(), "REMOVE_FRAGMENT", Toast.LENGTH_SHORT).show();
                return true;
            default:
                return super.onOptionsItemSelected(item);

        }
    }

    @Override
    public void onPrepareOptionsMenu(Menu menu) {
        super.onPrepareOptionsMenu(menu);
        MenuItem item = menu.findItem(R.id.add_item_fragment);
        if (isRefresh) {
            item.setTitle("REFRESHED_FRAGMENT");
        } else {
            item.setTitle(item.getTitle());
        }
    }
}

File: android_kotlin/menu_demo/app/src/main/java/com/heyangpeng/menu_demo/FirstFragment.kt
package com.heyangpeng.menu_demo

import android.content.ClipData
import android.os.Bundle
import android.view.*
import android.widget.Toast
import androidx.fragment.app.Fragment

class FirstFragment : Fragment() {
    private var isRefresh: Boolean = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.activity_fragment, container, false)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.menu_fragment, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.add_item_fragment -> {
                isRefresh = true
                activity?.invalidateOptionsMenu()
                Toast.makeText(context, "REFRESH_FRAGMENT", Toast.LENGTH_SHORT).show()
            }
            R.id.remove_item_fragment -> Toast.makeText(
                context,
                "REMOVE_FRAGMENT",
                Toast.LENGTH_SHORT
            ).show()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onPrepareOptionsMenu(menu: Menu) {
        super.onPrepareOptionsMenu(menu)
        val item = menu.findItem(R.id.add_item_fragment)
        if (isRefresh) item.title = "REFRESHED_FRAGMENT"
    }
}